internal/exec/vendor_component_utils.go

```diff
@@ -352,72 +309,53 @@ func ExecuteComponentVendorInternal(
-				uri,
-				component,
-				path.Join(componentPath, mixin.Filename),
-			))
-
-			if !dryRun {
-				err = os.RemoveAll(tempDir)
-				if err != nil {
-					return err
-				}
-
-				// Download the mixin into the temp file
-				if !useOciScheme {
-					client := &getter.Client{
-						Ctx:  context.Background(),
-						Dst:  path.Join(tempDir, mixin.Filename),
-						Src:  uri,
-						Mode: getter.ClientModeFile,
-					}
-
-					if err = client.Get(); err != nil {
-						return err
-					}
-				} else {
-					// Download the Image from the OCI-compatible registry, extract the layers from the tarball, and write to the destination directory
-					err = processOciImage(cliConfig, uri, tempDir)
-					if err != nil {
-						return err
-					}
-				}
-
-				// Copy from the temp folder to the destination folder
-				copyOptions := cp.Options{
-					// Preserve the atime and the mtime of the entries
-					PreserveTimes: false,
-
-					// Preserve the uid and the gid of all entries
-					PreserveOwner: false,
-
-					// OnSymlink specifies what to do on symlink
-					// Override the destination file if it already exists
-					// Prevent the error:
-					// symlink components/terraform/mixins/context.tf components/terraform/infra/vpc-flow-logs-bucket/context.tf: file exists
-					OnSymlink: func(src string) cp.SymlinkAction {
-						return cp.Deep
-					},
-				}
-
-				if err = cp.Copy(tempDir, componentPath, copyOptions); err != nil {
-					return err
-				}
+			pkg := pkgComponentVendor{
+				uri:                 uri,
+				pkgType:             pType,
+				name:                "mixin " + uri,
+				sourceIsLocalFile:   sourceIsLocalFile,
+				IsMixins:            true,
+				vendorComponentSpec: vendorComponentSpec,
+				version:             mixin.Version,
+				componentPath:       componentPath,
+				mixinFilename:       mixin.Filename,
 			}
+
+			packages = append(packages, pkg)
+		}
+	}
+	// Run TUI to process packages
+	if len(packages) > 0 {
+		model, err := newModelComponentVendorInternal(packages, dryRun, cliConfig)
+		if err != nil {
+			return fmt.Errorf("error initializing model: %v", err)
+		}
+		var opts []tea.ProgramOption
+		// Disable TUI if no TTY support is available
+		if !CheckTTYSupport() {
+			opts = []tea.ProgramOption{tea.WithoutRenderer()}
+		}
+		if _, err := tea.NewProgram(model, opts...).Run(); err != nil {
+			return fmt.Errorf("running download error: %w", err)
 		}
 	}
-
 	return nil
 }
 
-// ExecuteStackVendorInternal executes the command to vendor an Atmos stack
-// TODO: implement this
-func ExecuteStackVendorInternal(
-	stack string,
-	dryRun bool,
-) error {
-	return fmt.Errorf("command 'atmos vendor pull --stack <stack>' is not supported yet")
+// CheckTTYSupport checks if both stdin  support TTY.
+func CheckTTYSupport() bool {
+
+	// Check for standard TTY support on stdin
+	stdinTTY := isatty.IsTerminal(os.Stdin.Fd())
+
+	// Return true if either standard TTY
+	return stdinTTY
```

Comment:
I feel like we're conflating TTYs and interactivy TTYs. A TTY can exist without being interactive.

e.g. when you use docker there an interactive TTY:
```
docker -it
```

And there's and interactive mode: (that accepts stdin)
```
docker -i
```

And there's and TTY that does not accept stdin input
```
docker -i
```

If vendoring, for example, does not prompt the user, we should not require interactive mode to display the TUI things.

Reply:
@coderabbitai any tips?